apps/www/components/Pledge/Submit.js

```diff
@@ -172,6 +236,110 @@ class Submit extends Component {
         ref && ref.getWrappedInstance ? ref.getWrappedInstance() : ref
     }
   }
+
+  isStripeWalletPayment() {
+    return this.props.selectedPaymentMethod?.startsWith('STRIPE-WALLET')
+  }
+
+  componentDidUpdate() {
+    const { t, setSelectedPaymentMethod } = this.props
+    // Skip if loading
+    if (this.props.paymentRequest.status === PaymentRequestStatus.LOADING) {
+      return
+    }
+
+    // Delay creation of a payment-request, which requires Stripe to be loaded,
+    // until a wallet payment-method is selected
+    const isUninitializedStripeWallet =
+      this.isStripeWalletPayment() &&
+      this.props.paymentRequest.status === PaymentRequestStatus.IDLE
+
+    // Since we don't want to always load all Payment Providers data,
+    // we must reinitialize when switching to a different wallet
+    const isOutdatedStripeWallet =
+      this.isStripeWalletPayment() &&
+      this.props.paymentRequest.status !== PaymentRequestStatus.IDLE &&
+      this.props.paymentRequest.usedWallet !== this.props.selectedPaymentMethod
+
+    const isUnavailableAndStripeWalletIsSelected =
+      this.isStripeWalletPayment() &&
+      this.props.paymentRequest.status === PaymentRequestStatus.UNAVAILABLE
+
+    if (isUninitializedStripeWallet || isOutdatedStripeWallet) {
+      const selectedPaymentMethod = this.props.selectedPaymentMethod
+
+      // Create payment-request
+      this.props.paymentRequest
+        .initialize(selectedPaymentMethod)
+        .then((status) => {
+          if (status !== PaymentRequestStatus.UNAVAILABLE) {
+            this.setState(() => ({
+              loading: false,
+              walletError: null,
+            }))
+          } else {
+            const isGooglePayPayment =
+              this.props.selectedPaymentMethod ===
+              WalletPaymentMethod.GOOGLE_PAY
+
+            if (isGooglePayPayment) {
+              setSelectedPaymentMethod('STRIPE')
+            }
+            this.setState((prevState) => ({
+              ...prevState,
+              loading: false,
+              walletError: !isGooglePayPayment
+                ? t('account/pledges/payment/methods/unavailable')
+                : null,
+              stripeError: isGooglePayPayment
+                ? t('account/pledges/payment/methods/google-pay/unavailable')
+                : null,
+            }))
```

Comment:
IMO this logic could be made generic, for the odd case where is happens with apple pay

Reply:
Done, now works for both Apple and Google Pay (also works if other Wallets are added)